pkg/api/handlers/slack.go

```diff
@@ -0,0 +1,154 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	gatewayTypes "github.com/obot-platform/obot/pkg/gateway/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	slackOAuthURL = "https://slack.com/oauth/v2/authorize"
+)
+
+func (p *ProjectsHandler) Configure(req api.Context) error {
+	thread, err := getThreadForScope(req)
+	if err != nil {
+		return err
+	}
+
+	var (
+		input struct {
+			AppID         string `json:"appID"`
+			SigningSecret string `json:"signingSecret"`
+			ClientID      string `json:"clientID"`
+			ClientSecret  string `json:"clientSecret"`
+		}
+	)
+
+	if err := req.Read(&input); err != nil {
+		return err
+	}
+
+	var threads v1.ThreadList
+	if err := req.List(&threads, &client.ListOptions{
```

Comment:
I would actually create a real resource called SlackReciever.  Don't just embed all the info elsewhere. This way you can do this check but also not polute the other types like thread with this information that's not fully needed.  Then also some of this logic for create/update can be put in a controller.  The oauthapp can then be created by the controller. If the oauthapp has a spec.threadname field then you don't need to add the oauthapps field to the thread because you can just list for oauth apps associated to the thread.

Reply:
Changed to have a new resource type `SlackReciever`